week3/src/App.js

```diff
@@ -0,0 +1,130 @@
+import {useState} from 'react'
+
+function Square({value, onSquareClick})
+{
+  return (
+  <button className="square" onClick={onSquareClick}>{value}</button>
+  )
+}
+
+function Board({xIsNext, squares, onPlay})
+{
+  function handleClick(index)
+  {
+    if(squares[index] || calculateWinner(squares))
+    {
+      return
+    }
+    const nextSquares = squares.slice()
+    if(xIsNext)
+    {
+      nextSquares[index] = 'X'
+    }
+    else
+    {
+      nextSquares[index] = 'O'
+    }
+    onPlay(nextSquares)
+  }
+
+  const winner = calculateWinner(squares)
+  let status
+  if(winner)
+  {
+    status = "Winner: " + winner
+  }
+  else
+  {
+    status = "Next Player: " + (xIsNext ? 'X' : 'O' )
+  }
+  return (
+  <>
+    <div className="status">{status}</div>
+    <div className="board-row">
+      <Square value={squares[0]} onSquareClick={() => handleClick(0)}/>
+      <Square value={squares[1]} onSquareClick={() => handleClick(1)}/>
+      <Square value={squares[2]} onSquareClick={() => handleClick(2)}/>
+    </div>
+    <div className="board-row">
+      <Square value={squares[3]} onSquareClick={() => handleClick(3)}/>
+      <Square value={squares[4]} onSquareClick={() => handleClick(4)}/>
+      <Square value={squares[5]} onSquareClick={() => handleClick(5)}/>
+    </div> 
+    <div className="board-row">
+      <Square value={squares[6]} onSquareClick={() => handleClick(6)}/>
+      <Square value={squares[7]} onSquareClick={() => handleClick(7)}/>
+      <Square value={squares[8]} onSquareClick={() => handleClick(8)}/>
+    </div>
+    
+  </>
+  )
+}
+
+export default function Game()
+{
+  const [history, sethistory] = useState([Array(9).fill(null)])
+  const [currentMove, setCurrentMove] = useState(0)
+  const xIsNext = currentMove % 2 === 0
+  const currentSquares = history[currentMove]
+
+  function handlePlay(nextSquares)
+  {
+    const nextHistory = [...history.slice(0, currentMove + 1), nextSquares]
+    sethistory(nextHistory)
+    setCurrentMove(nextHistory.length - 1);
+  }
+
+  function jumpTo(move)
+  {
+    setCurrentMove(move)
+  }
+
+  const moves = history.map((squares, move) => {
+    let description
+    if(move > 0)
+    {
+      description = 'Go to move #' + move
+    }
+    else
+    {
+      description = 'Go to game start'
+    }
+
+    return (
+      <li key={move}>
+        <button onClick={() => jumpTo(move)}>{description}</button>
+      </li>
+    )
+  })
+  return (
+    <div className="game">
+      <div className="game-board">
+        <Board xIsNext={xIsNext} squares={currentSquares} onPlay={handlePlay}/>
+      </div>
+      <div className="game-info">
+      {moves}
+      </div>
+    </div>
+  )
+}
+
+function calculateWinner(squares) {
```

Comment:
create a file names `utile/calculate-winner.js` and move the calcaulteWinner function to the new file.